File: mongodb_related/make_review_file.py
from pymongo import MongoClient

url = "mongodb+srv://mark:WNQmnmMW1Eob4gFi@cluster0.gvyaavk.mongodb.net/myFirstDatabase?authMechanism=DEFAULT"
client = MongoClient(url)

db = client["gp"]
collection = db["map"]

# get all the reviews
store = collection.find({"reviews": {"$exists": True}})

# store reviews in a file
count = 0
with open("reviews.txt", "w") as f:
    tmpString = ""
    for s in store:
        for review in s["reviews"]:
            tmpString += review + " "
        
        tmpString.replace("\r\n", " ")
        tmpString.replace("\n", " ")
        ' '.join(tmpString.split())
        tmpString = repr(tmpString).replace('\\r\\n', ' ').replace('\\n', ' ').replace('\\', '')
        tmpString = tmpString[1:-1]

        f.write(s["place_id"] + "," + tmpString + "\n")
        count += 1
        # print(count, " ", s["place_id"])
        # if count > 0:
        #     break

File: mongodb_related/make_review_file.js
var MongoClient = require('mongodb').MongoClient;
 
const main = async () => {
    const client = await MongoClient.connect("mongodb+srv://mark:WNQmnmMW1Eob4gFi@cluster0.gvyaavk.mongodb.net/myFirstDatabase?authMechanism=DEFAULT", { useUnifiedTopology: true })
    const db = await client.db("gp")
    const collection = await db.collection("map")
    const result = await collection.find({}).toArray()

    const fs = require('fs');
    const path = require('path');
    const filePath = path.join(__dirname, 'review_js.json');

    for(let i = 0; i < result.length; i++){
        const obj = result[i];
        const reviews = obj.reviews;
        let tmpString = "";
        for(let j = 0; j < reviews.length; j++){
            const review = reviews[j];
            tmpString += review + " "
        }
        tmpString = tmpString.replace(/\n/g,' ');
        collection.updateOne({_id: obj._id}, {$set: {concated_review: tmpString}})
        // fs.writeFile(filePath, (obj.place_id + " " + tmpString));
        console.log(obj.place_id + " finished!");
    }
}

main().catch(console.error);